docker-storage-setup.sh

```diff
@@ -839,6 +853,123 @@ get_existing_storage_driver() {
   return 1
 }
 
+docker_root_volume_exists() {
+    lvs $VG/$DOCKER_ROOT_LV_NAME > /dev/null 2>&1 && return 0
+    return 1
+}
+
+# Check if docker root volume is already mounted.
+# This return 0 if volume is already mounted on $DOCKER_ROOT_DIR. Returns 1
+# if volumes needs mounting. On all other error conditions it exits.
+
+docker_root_lv_mounted() {
+  local mounts target
+  if ! mounts=$(findmnt -n --first-only --source /dev/$VG/$DOCKER_ROOT_LV_NAME); then
+    return 1
+  fi
+  [ -z "$mounts" ] && return 1
+  # If root volume is mounted, it should be mounted on $DOCKER_ROOT_DIR
+  # otherwise it is an error.
+  target=`echo $mounts | cut -d " " -f1`
+  [ "$target" == "$DOCKER_ROOT_DIR" ] && return 0
+    Error "Docker root volume /dev/$VG/$DOCKER_ROOT_LV_NAME is mounted on $target instead of ${DOCKER_ROOT_DIR}. Unmount it and try again."
+    exit 1
+}
+
+# Mount docker root volume on $DOCKER_ROOT_DIR.
+# drop a systemd mount unit configuration file at /etc/systemd/system.
+# This would allow the $DOCKER_ROOT_DIR mountpoint to persist across reboots.
+mount_docker_root_volume() {
+  # filename must match $DOCKER_ROOT_DIR path. e.g if $DOCKER_ROOT_DIR is /var/lib/docker
+  # then filename will be var-lib-docker.mount
+  filename=$(echo ${DOCKER_ROOT_DIR}.mount|sed 's|/|-|g' | cut -b 2-)
+  cat <<EOF > /etc/systemd/system/$filename
+  [Unit]
+  Description=Mount docker-root-lv on docker root directory.
+
+  [Mount]
+  What=/dev/$VG/$DOCKER_ROOT_LV_NAME
+  Where=${DOCKER_ROOT_DIR}
+  Type=xfs
+  Options=defaults
+
+  [Install]
+  WantedBy=multi-user.target
```

Comment:
Should we use `WantedBy=local-fs.target` instead? And possibly add `Before=local-fs.target` as well. 

man `systemd.special` says systemd-fstab-generator automatically adds dependencies for entries in /etc/fstab file. Given this will not be in fstab file, that probably means we will have to add dependencies ourselves.